pkg/controller/migdirect/rsync.go

```diff
@@ -0,0 +1,827 @@
+							},
+							{
+								Name:      "stunnel-certs",
+								MountPath: "/etc/stunnel/certs",
+							},
+						},
+						SecurityContext: &corev1.SecurityContext{
+							Privileged:             &trueBool,
+							RunAsUser:              &runAsUser,
+							ReadOnlyRootFilesystem: &trueBool,
+						},
+					},
+				},
+			},
+		}
+		err = destClient.Create(context.TODO(), &transferPod)
+		if k8serror.IsAlreadyExists(err) {
+			t.Log.Info("Rsync transfer pod already exists on destination", "namespace", transferPod.Namespace)
+		} else if err != nil {
+			return err
+		}
+		t.Log.Info("Rsync transfer pod created", "name", transferPod.Name, "namespace", transferPod.Namespace)
+
+	}
+	return nil
+}
+
+func (t *Task) getPVCNamespaceMap() map[string][]string {
+	nsMap := map[string][]string{}
+	for _, pvc := range t.Owner.Spec.PersistentVolumeClaims {
+		if vols, exists := nsMap[pvc.Namespace]; exists {
+			vols = append(vols, pvc.Name)
+			nsMap[pvc.Namespace] = vols
+		} else {
+			nsMap[pvc.Namespace] = []string{pvc.Name}
+		}
+	}
+	return nsMap
+}
+
+func (t *Task) getRsyncRoute(namespace string) (string, error) {
+	// Get client for destination
+	destClient, err := t.getDestinationClient()
+	if err != nil {
+		return "", err
+	}
+	route := routev1.Route{}
+
+	key := types.NamespacedName{Name: "migdirect-rsync-transfer-route", Namespace: namespace}
+	err = destClient.Get(context.TODO(), key, &route)
+	if err != nil {
+		return "", err
+	}
+	return route.Spec.Host, nil
+}
+
+// Create rsync client pods
+func (t *Task) createRsyncClientPods() error {
+	// Get client for destination
+	srcClient, err := t.getSourceClient()
+	if err != nil {
+		return err
+	}
+
+	pvcMap := t.getPVCNamespaceMap()
+	for ns, vols := range pvcMap {
+		// Get stunnel svc IP
+		svc := corev1.Service{}
+		key := types.NamespacedName{Name: "migdirect-rsync-transfer-svc", Namespace: ns}
+		srcClient.Get(context.TODO(), key, &svc)
+		ip := svc.Spec.ClusterIP
+
+		trueBool := true
+		runAsUser := int64(0)
+
+		// Add PVC volume mounts
+		for _, vol := range vols {
+			volumes := []corev1.Volume{}
+			volumeMounts := []corev1.VolumeMount{}
+			containers := []corev1.Container{}
+			volumeMounts = append(volumeMounts, corev1.VolumeMount{
+				Name:      vol,
+				MountPath: fmt.Sprintf("/mnt/%s/%s", ns, vol),
+			})
+			volumes = append(volumes, corev1.Volume{
+				Name: vol,
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: vol,
+					},
+				},
+			})
+			containers = append(containers, corev1.Container{
+				Name:  fmt.Sprintf("rsync-client-%s", vol),
+				Image: "quay.io/konveyor/rsync-transfer:latest",
+				Env: []corev1.EnvVar{
+					{
+						Name:  "RSYNC_PASSWORD",
+						Value: "changeme",
+					},
+				},
+				Command: []string{"rsync", "-aPvvHh", "--delete", "--port", "2222", "--log-file", "/dev/stdout", fmt.Sprintf("/mnt/%s/%s/", ns, vol), fmt.Sprintf("rsync://root@%s/%s", ip, vol)},
```

Comment:
I was reading an article the other day that was making an argument for always using long-form CLI flags when scripting _for understandability_ instead of the compressed `-avPvvHh` flag sets that you would use to _save time when typing_ out a command. 

I know we're using this in pvc-migrate and probably want to maintain continuity so I'm not going to say we need to change it here, just an interesting thought I wanted to share.

The equivalent flag set would be:
```
-aPvvHh
--archive --verbose --progress --human-readable --hard-links --vv
```

Reply:
@alaypatel07 ^ please include this update with the progress fix